sklego/metrics.py

```diff
@@ -152,3 +154,36 @@ def impl(estimator, X, y_true):
         return score if not np.isnan(score) else 1
 
     return impl
+
+
+def subset_metric(subset_picker: Callable, metric: Callable):
+    r"""
+    Returns a method that applies the passed metric only to a specific subset. The subset picker
+    is a method that is passed the corresponding X and y_true and returns a one-dimensional
+    boolean vector where every element corresponds to a row in the data. Only the elements
+    with a True value are taken into account for the passed metric, representing a filter.
+
+    This allows users to have an easy approach to measuring metrics over different slices of
+    the population which can give insights into the model performance, either specifically for
+    fairness or in general.
+
+    Usage:
+    `subset_metric(lambda X, y_true: subset_method, metric_method)(clf, X, y)`
```

Comment:
Ok I will change the example. 

I included `y` so that you could use the method to also slice on outcomes. This way you could - for example - directly ask what the MSE is for predicted heart rates of women who fall in the high heart rate category.

Reply:
I kept in the `y` for now, processed all the other feedback and updated the PR. If this choice of including `y` is fine then I think it is ready.

Reply:
I think including `y` is a good idea. For example the `equal_opportunity_score` requires access to `y` and I could see more metrics having the same requirements.

Reply:
fair enough.